control-plane/csi-driver/src/bin/node/main_.rs

```diff
@@ -427,21 +442,35 @@ async fn check_ana_and_label_node(
 
 /// Validate that the grpc endpoint is valid.
 fn validate_endpoints(
-    grpc_endpoint: &str,
+    matches: &clap::ArgMatches,
     registration_enabled: bool,
 ) -> anyhow::Result<SocketAddr> {
-    let grpc_endpoint = if grpc_endpoint.contains(':') {
-        grpc_endpoint.to_string()
+    let grpc_endpoint_src = matches.value_source("grpc-endpoint");
+    let grpc_ip = matches.get_one::<IpAddr>("grpc-ip");
+    let grpc_port = matches.get_one::<u16>("grpc-port");
+    let grpc_endpoint_socket_addr = if grpc_endpoint_src
+        .unwrap_or(clap::parser::ValueSource::DefaultValue)
+        == clap::parser::ValueSource::DefaultValue
```

Comment:
What about making `grpc-endpoint` optional, and adding a clap conflicts with `grpc-ip` and `grpc-port` ? 
Would it be more intuitive?

Reply:
I added a `conflicts_with_all` for both the args. Does that help with using `matches` here somehow? I'm no expert when it comes to clap's builder, feel free to make a suggestion :)

Reply:
awesome, that's it, now it complains straight away:

```
❯ csi-node --node-name a --grpc-endpoint 127.0.0.1 --grpc-ip 127.0.0.1 --grpc-port 1080                                                                                                                                                  ─╯
error: the argument '--grpc-endpoint <ENDPOINT>' cannot be used with:
  --grpc-ip <GRPC_IP>
  --grpc-port <GRPC_PORT>
```

Also in addition, you can remove the default value from `grpc-endpoint`.

Reply:
done! simplified the validation code